test/e2e/framework/helper/validations/certificates.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	return nil
+}
+
+// ExpectValidCommonName checks if the issued certificate has the requested CN or one of the DNS SANs
+func ExpectValidCommonName(certificate *cmapi.Certificate, secret *corev1.Secret) error {
+	cert, err := pki.DecodeX509CertificateBytes(secret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return err
+	}
+
+	expectedCN := certificate.Spec.CommonName
+
+	if len(expectedCN) == 0 && len(cert.Subject.CommonName) > 0 {
+		// no CN is specified but our CA set one, checking if it is one of our DNS names
+		if !util.Contains(cert.DNSNames, cert.Subject.CommonName) {
+			return fmt.Errorf("Expected a common name for one of our DNSNames %v, but got a CN of %v", cert.DNSNames, cert.Subject.CommonName)
+		}
+	} else if expectedCN != cert.Subject.CommonName {
+		return fmt.Errorf("Expected a common name of %v, but got a CN of %v", expectedCN, cert.Subject.CommonName)
+	}
+
+	return nil
+}
+
+// ExpectValidNotAfterDate checks if the issued certificate matches the requested duration
+func ExpectValidNotAfterDate(certificate *cmapi.Certificate, secret *corev1.Secret) error {
+	cert, err := pki.DecodeX509CertificateBytes(secret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return err
+	}
+	if certificate.Status.NotAfter == nil {
+		return fmt.Errorf("No certificate expiration found for Certificate %q", certificate.Name)
+	}
+
+	if !cert.NotAfter.Equal(certificate.Status.NotAfter.Time) {
+		return fmt.Errorf("Expected certificate expiry date to be %v, but got %v", certificate.Status.NotAfter, cert.NotAfter)
+	}
+
+	return nil
+}
+
+func containsExtKeyUsage(s []x509.ExtKeyUsage, e x509.ExtKeyUsage) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+// ExpectKeyUsageExtKeyUsageServerAuth checks if the issued certificate has the extended key usage of server auth
+func ExpectKeyUsageExtKeyUsageServerAuth(certificate *cmapi.Certificate, secret *corev1.Secret) error {
+	cert, err := pki.DecodeX509CertificateBytes(secret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return err
+	}
+
+	if !containsExtKeyUsage(cert.ExtKeyUsage, x509.ExtKeyUsageServerAuth) {
+		return fmt.Errorf("Expected certificate to have ExtKeyUsageServerAuth, but got %v", cert.ExtKeyUsage)
+	}
+	return nil
+}
+
+// ExpectKeyUsageExtKeyUsageServerAuth checks if the issued certificate has the extended key usage of client auth
+func ExpectKeyUsageExtKeyUsageClientAuth(certificate *cmapi.Certificate, secret *corev1.Secret) error {
+	cert, err := pki.DecodeX509CertificateBytes(secret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return err
+	}
+
+	if !containsExtKeyUsage(cert.ExtKeyUsage, x509.ExtKeyUsageClientAuth) {
+		return fmt.Errorf("Expected certificate to have ExtKeyUsageClientAuth, but got %v", cert.ExtKeyUsage)
+	}
+	return nil
+}
+
+// ExpectKeyUsageKeyUsageKeyAgreement checks if the issued certificate has the key usage of key agreement
```

Comment:
I don't think this is used. Remove it?

```
richard   e2e-venafi-cloud  ~  projects  cert-manager  cert-manager  git grep ExpectKeyUsageKeyUsageKeyAgreement
test/e2e/framework/helper/validations/certificates.go:// ExpectKeyUsageKeyUsageKeyAgreement checks if the issued certificate has the key usage of key agreement
test/e2e/framework/helper/validations/certificates.go:func ExpectKeyUsageKeyUsageKeyAgreement(certificate *cmapi.Certificate, secret *corev1.Secret) error {

```

If we keep it, then add a few further comments explaining what this

Reply:
came out of a vault specific thing, removed it as it wasn't quite a useful check to do